### DISCUSSION TITLE
Solar Energy Produced today

### DISCUSSION THREAD

**Discussion by greenlava82** [alandtse/tesla#1073] [Ideas]
Hi!  I'm new to HA (and thus new to HA development) but I'm a software engineer with some python experience.  I'm interested to add on a new sensor to your integration that will query the API for calendar_history

/api/1/energy_sites/{site_id}/calendar_history

with kind=energy, end_date={today}, and period=day

Ideally, this will give some value representing the current day's aggregate production for the site, but at the moment I haven't tested this.  I'd love to help out with this, or even implement it entirely, but as a new HA dev I'm not 100% where to take my first bite.  If you'd be willing to give a quick overview of where you make the API call, where you receive that call back, and where you parse the values into sensor values I'd appreciate it.

**Comment by alandtse** [comment]
Take a look at https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/endpoints.json and submit PRs there. After they are accepted, you can bump here and create whatever sensors make sense. Please note there is an official Tesla integration so that may be the better place to start as this is at the tail end of development.
